### PR TITLE
cache rc support

### PR DESCRIPTION
**Pull request by remusb**
Hi @ncw 

I've added support for the RC module in cache and updated documentation. I'm opening as a PR to discuss if I can add something more in the RC. To me, it looks final as it has all the support I initially wanted to add too.

I've also updated the docs of rc with the params.

PS: This is on a new branch cause I wanted to rebase master.

### CODE REVIEW

**Comment by remusb** [comment]
Let's link the issue too: https://github.com/ncw/rclone/issues/2111

File: docs/content/cache.md [hunk 1]
```diff
@@ -242,6 +242,15 @@ which makes it think we're downloading the full file instead of small chunks.
...
 <span style="color:green">**cloud remote** -> **cache** -> **crypt**</span>
 
+### Cache and Remote Control (--rc) ###
+Cache supports the new `--rc` mode in rclone and can be remote controlled through the following end points:
+
+### rc cache/expire
+Purge a remote from the cache backend. Supports either a directory or a file.
+Params:
+  - **remote** = path to remote **(required)**
+  - **withData** = true/false to delete cached data (chunks) as well _(optional)_
```
Comment (seuffert) [inline docs/content/cache.md]: i think it would be good to mention that **_false_** is the standard

**Comment by ncw** [comment]
I think that looks good..

Do you think rc has enough features for your use?

Thanks for updating the docs :-)

Do you want to merge this for v1.40 which I'm planning for next weekend 18th March?  I'll need to tidy rc up a bit before then if so!

**Comment by remusb** [comment]
@ncw 1.4 sounds great. It's enough for what I wanted to do so yep. Let me know if I can help contribute more to the rc code.

I'll have to fix the SetIndent support in cmd rc, will push again this evening

File: docs/content/cache.md (hunk 1 above)
Comment (remusb) [reply docs/content/cache.md]: That makes sense. Thanks

**Comment by remusb** [comment]
@ncw this looks done. wanna have a final look on it or can I merge it?

**Comment by ncw** [comment]
Can you split out the go1.6 compatibility  fixes for rc out of the cache fixes in fc47c5999bdea553e3af792a77f83fb503388842

Also can you fixup the title of f2d0d15a1a28109bdbf1c639ffd0895ff66a8fca so it doesn't say FIXME WIP in it before you merge :-)

After that merge away!

**Comment by remusb** [comment]
@ncw ok, done. I'm waiting for the tests to finish and will merge

**Pull request merged**